bathbot/src/commands/osu/relax/profile.rs

```diff
@@ -0,0 +1,349 @@
+
+            None
+        }
+    };
+
+    let info_res = match info_res {
+        Ok(Some(info_res)) => info_res,
+        Ok(None) => {
+            return orig
+                .error(format!("User `{}` not found", user.username))
+                .await;
+        }
+        Err(err) => {
+            warn!(?err, "Failed to fetch user relax player info");
+
+            let _ = orig.error(GENERAL_ISSUE).await;
+
+            return Err(err);
+        }
+    };
+
+    let origin = MessageOrigin::new(orig.guild_id(), orig.channel_id());
+    let pagination = RelaxProfileArgs::new(user, discord_id, info_res, origin);
+
+    let graph = match relax_playcount_graph(&pagination) {
+        Ok(graph) => graph,
+        Err(err) => {
+            let _ = orig.error(GENERAL_ISSUE).await;
+
+            return Err(err.wrap_err("Failed to draw a relax playcount graph"));
+        }
+    };
+
+    let builder = MessageBuilder::new()
+        .embed(relax_profile_builder(pagination).unwrap())
+        .attachment("graph.png", graph);
+
+    orig.create_message(builder).await?;
+
+    Ok(())
+}
+
+pub struct RelaxProfileArgs {
+    user: CachedUser,
+    discord_id: Option<Id<UserMarker>>,
+    info: RelaxPlayersDataResponse,
+    origin: MessageOrigin,
+}
+
+impl RelaxProfileArgs {
+    pub fn new(
+        user: CachedUser,
+        discord_id: Option<Id<UserMarker>>,
+        info: RelaxPlayersDataResponse,
+        origin: MessageOrigin,
+    ) -> Self {
+        Self {
+            user,
+            discord_id,
+            info,
+            origin,
+        }
+    }
+}
+
+pub fn relax_profile_builder(args: RelaxProfileArgs) -> Result<EmbedBuilder> {
+    let stats = &args.info;
+    let mut description = "__**Relax user statistics".to_string();
+    if let Some(discord_id) = args.discord_id {
+        let _ = write!(description, "for <@{discord_id}>");
+    };
+
+    description.push_str(":**__");
+    let _ = writeln!(
+        description,
+        "\n
+        Accuracy: [`{acc:.2}%`]({origin} \"{acc}\") • \
+        Playcount: `{playcount}`",
+        origin = args.origin,
+        acc = stats.total_accuracy.unwrap_or_default(),
+        playcount = WithComma::new(stats.playcount)
+    );
+    let ss_grades = format!("{}{}", stats.count_ss, grade_emote(Grade::X));
+    let s_grades = format!("{}{}", stats.count_s, grade_emote(Grade::S));
+    let a_grades = format!("{}{}", stats.count_a, grade_emote(Grade::A));
+    let fields = fields![
+        "Count SS", ss_grades, true;
+        "Count S", s_grades, true;
+        "Count A", a_grades, true;
+    ];
+    let embed = EmbedBuilder::new()
+        .author(relax_author_builder(&args))
+        .description(description)
+        .fields(fields)
+        .image(attachment("graph.png"))
+        .thumbnail(args.user.avatar_url.as_ref())
+        .footer(relax_footer_builder(&args));
+
+    Ok(embed)
+}
+
+fn relax_author_builder(args: &RelaxProfileArgs) -> AuthorBuilder {
```

Comment:
Can this method be refactored together with the author builder for the top subcommand? If not that's fine, just checking in